### DISCUSSION TITLE
Need help converting command line to C#

### DISCUSSION THREAD

**Discussion by cclops** [ImageMagick/ImageMagick#2494] [Help]
I have no experience using the command line.  I have only used Magick.NET 

I need to create a wood grain frame around an image.  I found this, but I don't know how to convert it over to C#.  I don't even know where to begin. Any help would be greatly appreciated

```
convert thumbnail.gif                -write mpr:image    +delete \
          goldthin_top.png             -write mpr:edge_top +delete \
          goldthin_btm.png -rotate 180 -write mpr:edge_btm +delete \
          \
          mpr:image -alpha set -bordercolor none \
          -compose Dst -frame 25x25+25  -compose over \
          \
          -tile mpr:edge_btm \
          -transverse -draw 'color 1,0 floodfill' \
          -transpose  -draw 'color 1,0 floodfill' \
          -tile mpr:edge_top \
          -transverse -draw 'color 1,0 floodfill' \
          -transpose  -draw 'color 1,0 floodfill' \
          \
          mpr:image -gravity center -composite    frame_gold.png
```

**Comment by fmw42** [comment]
I think you are missing some end of line  escapes \\.  I have corrected your code. Next time highlight the code and then press  the <> button to make sure the code is shown as pasted. 

Please provide a link to where you found that code.

**Comment by cclops** [comment]
Apologies

http://www.imagemagick.org/Usage/thumbnails/#frame

**Reply by fmw42** [reply]
More precisely at https://www.imagemagick.org/Usage/thumbnails/#frame_joints

**Reply by cclops** [reply]
Yes, thank you.

**Reply by fmw42** [reply]
Sorry, I do not code for Magick.Net. So you will have to wait for the Magick.Net developer to respond.

**Reply by cclops** [reply]
Thanks for helping me format my question, first time posting here.

**Comment by cclops** [comment]
I think I might have picked something more complex than I need. 

If I could get help translating this for Magick.Net from the same page, it would be most appreciated. 

https://www.imagemagick.org/Usage/thumbnails/#frame_joints

```
convert thumbnail.gif -alpha set -bordercolor none \
          -compose Dst_Out -frame 15x15+15  frame_template.gif

  convert frame_template.gif \
          -tile blackthin_top.gif   -draw 'color 1,0 floodfill' \
          -tile-offset +0+105 -tile blackthin_btm.gif \
                                       -draw 'color 15,105 floodfill' \
          -transpose \
          -tile blackthin_top.gif      -draw 'color 1,0 floodfill' \
          -tile-offset +0+135 -tile blackthin_btm.gif \
                                       -draw 'color 15,135 floodfill' \
          -transpose \
          -gravity center thumbnail.gif -composite frame_filled.gif

```

**Reply by fmw42** [reply]
I had to fix your post again. You did not select the code and press the <> button

**Reply by cclops** [reply]
Apologies, and thank you.
